### PR TITLE
Update the chrome extension to query for content by URI

### PR DESCRIPTION
**Pull request by shroudofthurin**
[COPILOT-5449](https://cnissues.atlassian.net/browse/COPILOT-5449)

### CODE REVIEW

File: background.js [hunk 1]
```diff
@@ -26,6 +25,62 @@ function getBrandFromHostname(hostname) {
...
+  })
+  .catch(function (err) {
+    chrome.browserAction.setBadgeText({text: '!', tabId: tab.id});
+    chrome.browserAction.setTitle({title: 'Error connecting to Copilot', tabId: tab.id});
+  });
+}
+
+function fetchCopilotData(path) {
+  return new Promise(function (resolve, reject) {
+    fetch(`https://copilot.aws.conde.io/${path}`, {credentials: 'include', redirect: 'manual'})
```
Comment (dadeg) [inline background.js]: Is there a need for supporting CI/stg? This design only supports production.

File: background.js [hunk 2]
```diff
@@ -26,6 +25,62 @@ function getBrandFromHostname(hostname) {
...
+    if (authObj && authObj.brands.indexOf(brand.code) > -1) {
+      return setBrandCookie(brand.code);
+    }
+    Promise.reject(new Error('User does not have access to brand'));
+  })
+  .then(searchCopilotByURI(encodeURIComponent(path)))
+  .then(function(data) {
+    if (data.hits.total === 1) {
+      let hit = data.hits.hits[0];
+      let url = `https://copilot.aws.conde.io/${brand.code}/${hit._source.meta.collectionName}/${hit._id}`;
```
Comment (dadeg) [inline background.js]: Is it possible to get the URL from the `_links.self` property of the hit? That may be better than generating it manually here.

File: inject.js [hunk 3]
```diff
@@ -1,15 +1,2 @@
...
-  window.dispatchEvent(event);
-}, 500);`;
-
-(document.head || document.documentElement).appendChild(script);
-
-window.addEventListener("cpScriptLoaded", function (e) {
-  digitalData = e.detail.digitalData;
-  chrome.runtime.sendMessage({ digitalData: digitalData });
-});
+chrome.runtime.sendMessage({greeting: "hello"});
```
Comment (dadeg) [inline inject.js]: is this needed?

File: background.js (hunk 2 above)
Comment (vivekvasani) [inline background.js]: I believe that link has the api domain in the URI, where we want the copilot app domain

File: background.js (hunk 1 above)
Comment (vivekvasani) [inline background.js]: maybe in the future, but unnecessary for now

File: inject.js (hunk 3 above)
Comment (shroudofthurin) [inline inject.js]: the greeting portion of `chrome.runtime.sendMessage({greeting: "hello"});` is not needed, but the `sendMessage` is needed to trigger the functions in `background.js`

File: background.js (hunk 2 above)
Comment (vivekvasani) [inline background.js]: on second thought, maybe that is a good idea.

1. grab the link from `_links`,
2. convert to URL object,
3. set the `hostname` to `copilot.aws.conde.io`

requires more steps, but could be better.

Comment (vivekvasani) [inline background.js]: I'm not really sure, I'll let you decide, @shroudofthurin

Comment (shroudofthurin) [inline background.js]: so `_links.self` has the correct copilot form link?

Comment (vivekvasani) [inline background.js]: i just inspected... that won't work. 

it's not only a matter of the hostnames being different but also adding the brand code in the path

Comment (shroudofthurin) [inline background.js]: @dadeg @vivekvasani `_links.self` does not provide you with the correct link. it provides you a link to the search result page. so, no it's not possible to get the URL from the `_link.self` property

**Pull request merged**